### PR TITLE
superlu-dist: fix case for cce fortran compiler

### PR DESCRIPTION
**Pull request by balos1**
cc @balay

### CODE REVIEW

File: var/spack/repos/builtin/packages/superlu-dist/package.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ class SuperluDist(CMakePackage, CudaPackage):
 
     patch('xl-611.patch', when='@:6.1.1 %xl')
     patch('xl-611.patch', when='@:6.1.1 %xl_r')
+    patch('superlu-cray-ftn-case.patch', when='%cce')
```
Comment (balay) [inline var/spack/repos/builtin/packages/superlu-dist/package.py]: @xiaoyeli @pghysels - perhaps this fix can go into superlu-dist sources.

And in spack - limit this patch to version 7.1.1 and lower?

Comment (pghysels) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: Yes, it should go in SuperLU-dist sources.
I fixed the same issue in STRUMPACK last week (too late for 6.1.0) https://github.com/pghysels/STRUMPACK/commit/b6a0fb02db57453fa04cef24ef29d122e5b8c181
(I initially tried to install *.mod using a wildcard, but that also installs empty folders, which led to infinite recursion when doing an in-source build: https://github.com/pghysels/STRUMPACK/commit/390835f92378466cc9c248680f4195056777eb8f

Comment (balay) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: for now - perhaps:

```suggestion
    patch('superlu-cray-ftn-case.patch', when='@:7.1.1 %cce')
```

**Comment by xiaoyeli** [comment]
@balay @pghysels  Do I need to do anything?  -Sherry

**Comment by pghysels** [comment]
@xiaoyeli I will let you know

**Comment by xiaoyeli** [comment]
@ balay:
I applied a cmake patch proposed by @pghysels.  It is now in Master,  will go into next release.
Your current resolution for 7.1.1 is fine.

**Review by balay — APPROVED** [review]

**Pull request merged**